### PR TITLE
Add support for using active FTP mode

### PR DESCRIPTION
**Pull request by rtblackwell**

### CODE REVIEW

File: data_pipeline_api/registry/common.py [hunk 1]
```diff
@@ -147,7 +147,8 @@ def get_remote_filesystem_and_path(
         return LocalFileSystem(**storage_options), uri.as_posix()
     elif protocol in {"http", "https", "s3"}:
         # storage_options are parameters passed to request
-        uri = urllib.parse.urljoin(uri, path)
+        path = urllib.parse.quote(path)
+        uri = '/'.join(s.strip('/') for s in [uri, path])
```
Comment (mrow84) [inline data_pipeline_api/registry/common.py:151]: Why are you no longer using `urljoin`? Could you use `urljoin` and then `quote`, or vice versa?

File: data_pipeline_api/registry/common.py [hunk 2]
```diff
@@ -156,7 +157,13 @@ def get_remote_filesystem_and_path(
         password = storage_options.pop("password", None) or inferred_options.get("password")
         uri = (Path(inferred_options["path"]) / Path(path)).as_posix()
         fs_class = FTPFileSystem if protocol == "ftp" else SFTPFileSystem
-        return fs_class(host=inferred_options["host"], username=username, password=password, **storage_options), uri
+        fs = fs_class(host=inferred_options["host"], username=username, password=password, **storage_options)
+        if protocol == "ftp":
+            try:
+                fs.ftp.dir()
+            except TimeoutError:
+                fs.ftp.set_pasv(False)
```
Comment (mrow84) [inline data_pipeline_api/registry/common.py:165]: I wonder if it is worth attempting the `dir` again after changing this?

**Review by mrow84 — APPROVED** [review]

File: data_pipeline_api/registry/common.py (hunk 1 above)
Comment (rtblackwell) [reply data_pipeline_api/registry/common.py:151]: urljoin is opinionated in ways that were not helpful to me, when I just want to be able to assume that you can do a + b and make sure there's a slash

File: data_pipeline_api/registry/common.py (hunk 2 above)
Comment (rtblackwell) [reply data_pipeline_api/registry/common.py:165]: I felt that that would be captured immediately on any use

**Pull request merged**